lib/tests/streamlit/widgets_test.py

```diff
@@ -111,6 +111,10 @@ def test_call_callbacks(self):
         _create_widget("int", states).int_value = 123
 
         widgets = WidgetStateManager()
+        # The state must be set twice, because the callback check expects the
+        # new state to have been set, so the old state will be stored and used
+        # for the callback checks.
+        widgets.set_state(states)
         widgets.set_state(states)
```

Comment:
The second `set_state` can be done on line 124 (before the `call_callbacks`) right? That's essentially why it's twice?

Reply:
Yes, it just needs to happen before `call_callbacks`. I'll move it to make it clearer.